[Feature] Add cache to disk

antonioloison opened the pull request:
resolves #3092. 

Added disk cache using the `diskcache` module.

Please check the issue for more context.

antonioloison committed:
build(caching.py) - add disk cache object

antonioloison committed:
test(test_caching.py): add disk cache test when using completion

ishaan-jaff commented on litellm/caching.py:
```diff
@@ -15,6 +15,7 @@
 from litellm._logging import verbose_logger
 from litellm.types.services import ServiceLoggerPayload, ServiceTypes
 import traceback
+import diskcache as dc
```
We don't want this impacting users on litellm proxy server

ishaan-jaff replied:
since this is an optional dependency it should only be imported in the class

antonioloison replied:
I understand, I moved it to the class.
I removed the diskcache dependency from the `pyproject.toml` file and added the install command in the docs in the same way as the `s3-cache`.

ishaan-jaff reviewed (changes requested):
left review, can you please add this to docs too. Thanks

antonioloison committed:
build(caching.py): add disk option for cache

antonioloison committed:
chore(pyproject.toml): add diskcache as extra

antonioloison committed:
test(test_caching.py): use mock_response in disk cache test

antonioloison committed:
chore: add diskcache as extra dependency

antonioloison committed:
build(caching.py): move diskcache import inside class and add cache_dir argument to Cache

antonioloison committed:
docs: add disk cache doc and update cache arguments

antonioloison commented on litellm/tests/test_caching.py:
```diff
@@ -875,6 +875,66 @@ async def test_redis_cache_acompletion_stream_bedrock():
+        type="disk",
+    )
+    print("test2 for Redis Caching - non streaming")
+    response1 = completion(
```
done, you can resolve the conversation if you think it is correct

antonioloison committed:
deps: remove diskcache from dependencies and add install in docs

antonioloison commented:
> left review, can you please add this to docs too. Thanks

I added the doc for the disk cache and corrected the arguments of the `add_cache`, `update_cache`... functions.
I also changed the name of the file `redis_cache.md` to `all_caches.md` as it no longer only concerns the redis cache.
